cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -486,36 +485,40 @@ private RestartPlan restartPlan(int podId, Pod pod, RestartContext restartContex
         }
 
         if (!needsRestart && allowReconfiguration) {
-            LOGGER.traceCr(reconciliation, "Broker {}: description {}", podId, brokerConfig);
-            diff = new KafkaBrokerConfigurationDiff(reconciliation, brokerConfig, kafkaConfig, kafkaVersion, podId);
-            loggingDiff = logging(podId);
+            LOGGER.traceCr(reconciliation, "Broker {}: description {}", podName, brokerConfig);
+            diff = new KafkaBrokerConfigurationDiff(reconciliation, brokerConfig, kafkaConfig, kafkaVersion, getPodIndexFromPodName(podName));
+            loggingDiff = logging(podName);
             if (diff.getDiffSize() > 0) {
                 if (diff.canBeUpdatedDynamically()) {
-                    LOGGER.debugCr(reconciliation, "Pod {} needs to be reconfigured.", podId);
+                    LOGGER.debugCr(reconciliation, "Pod {} needs to be reconfigured.", podName);
                     needsReconfig = true;
                 } else {
-                    LOGGER.debugCr(reconciliation, "Pod {} needs to be restarted, because reconfiguration cannot be done dynamically", podId);
+                    LOGGER.debugCr(reconciliation, "Pod {} needs to be restarted, because reconfiguration cannot be done dynamically", podName);
                     needsRestart = true;
                 }
             }
 
             if (loggingDiff.getDiffSize() > 0) {
-                LOGGER.debugCr(reconciliation, "Pod {} logging needs to be reconfigured.", podId);
+                LOGGER.debugCr(reconciliation, "Pod {} logging needs to be reconfigured.", podName);
                 needsReconfig = true;
             }
         } else if (needsRestart) {
-            LOGGER.infoCr(reconciliation, "Pod {} needs to be restarted. Reason: {}", podId, reasonToRestartPod);
+            LOGGER.infoCr(reconciliation, "Pod {} needs to be restarted. Reason: {}", podName, reasonToRestartPod);
         }
         return new RestartPlan(needsRestart, needsReconfig, podStuck, diff, loggingDiff);
     }
 
+    int getPodIndexFromPodName(String podName)  {
```

Comment:
No need for the `get`! It's a function; it's allowed to return a value without needing to make a song and dance about it by prefixing with `get`.

`indexOfPod`?